google_guest_agent/metadata_test.go

```diff
@@ -100,3 +100,29 @@ func TestBlockProjectKeys(t *testing.T) {
 		}
 	}
 }
+
+func TestGetNonExpiredKeys(t *testing.T) {
+	tests := []struct {
+		json string
+		res  map[string][]string
+	}{
+		{
+			`{"instance": {"attributes": {"ssh-keys": "malformed-ssh-keys"}}}`,
+			map[string][]string{},
+		},
+		{
+			`{"instance": {"attributes": {"sshKeys": ":malformed-ssh-keys"}}}`,
+			map[string][]string{},
+		},
+	}
+	for _, test := range tests {
+		var md metadata
+		if err := json.Unmarshal([]byte(test.json), &md); err != nil {
+			t.Errorf("failed to unmarshal JSON: %v", err)
+		}
+		mdKeyMap := getNonExpiredKeys(md.Instance.Attributes.SSHKeys)
+		if !reflect.DeepEqual(mdKeyMap, test.res) {
```

Comment:
add the specific comparisons you want to make, rather than using reflect

Reply:
directly compare map and slice and remove reflect